src/time/rfc1123.cc

```diff
@@ -44,7 +44,13 @@ make_month(const char *s)
     char month[3];
 
     month[0] = xtoupper(*s);
+    if (!month[0])
+        return -1; // protects *(s + 1) below
+
     month[1] = xtolower(*(s + 1));
+    if (!month[1])
+        return -1; // protects *(s + 2) below
+
     month[2] = xtolower(*(s + 2));
```

Comment:
This is not how we should write code these days, but improving this code would require rewriting the whole function (and more), which is not a good idea for this minimal fix. AFAICT from looking at optimized assembly code, this fix has reasonable performance impact; I cannot make it faster without expanding the scope of these changes. The entire make_month() function (and several of its siblings!) should be replaced with safer, optimized versions anyway.

Reply:
I agree with all you write. Maybe I can look into these
after html_quote is done

Reply:
Agreed. Fix first polish later.